### PR TITLE
[routing-utils] Update header replacing to handle more cases

### PR DESCRIPTION
**Pull request by ijjk**
This adds handling for more cases while updating header values to make sure to escape any characters that could break compiling with `path-to-regexp`

x-ref: https://github.com/vercel/next.js/pull/15592

### CODE REVIEW

File: packages/now-routing-utils/test/superstatic.spec.js [hunk 1]
```diff
@@ -534,12 +607,14 @@ test('convertHeaders', () => {
...
     ['404.html'],
     ['/blog/first-post', '/blog/another/one'],
+    ['/like/params/first', '/like/params/second'],
   ];
 
   const mustNotMatch = [
     ['hello/file.jpg', 'hello/font-css', 'dir/arial.font-css'],
     ['403.html', '500.html'],
     ['/blogg', '/random'],
+    ['/non-match', '/like/params'],
```
Comment (styfle) [inline packages/now-routing-utils/test/superstatic.spec.js]: What about `/like/params/`? Should that match or not?

File: packages/now-routing-utils/src/superstatic.ts [hunk 2]
```diff
@@ -218,18 +218,42 @@ function replaceSegments(
...
-      return compiler(indexes);
-    })
-    .join('?');
+
+  for (const key of Object.keys(indexes)) {
+    if (value.includes(`:${key}`)) {
+      value = value
+        .replace(
+          new RegExp(`:${key}\\*`, 'g'),
+          `:${key}--ESCAPED_PARAM_ASTERISKS`
```
Comment (styfle) [inline packages/now-routing-utils/src/superstatic.ts]: Nit pick: this is plural while the others are singular. Perhaps remove the `S`?

Comment (ijjk) [reply packages/now-routing-utils/src/superstatic.ts]: Updated

File: packages/now-routing-utils/test/superstatic.spec.js (hunk 1 above)
Comment (ijjk) [reply packages/now-routing-utils/test/superstatic.spec.js]: It should not match since the param isn't optional, updated test to reflect this also

**Review by styfle — APPROVED** [review]
Beautiful, thanks! 🦄

**Review by Timer — APPROVED** [review]
LGTM, as we landed the same code in Next.js! Thanks!

**Comment by styfle** [comment]
@ijjk Don't forget to update the API repo after publishing canary 👍

**Pull request merged**